packages/core/src/__tests__/ceramic-feed.test.ts

```diff
@@ -36,134 +57,110 @@ describe('Ceramic feed', () => {
-      await TileDocument.load(ceramic2, stream1.id)
-      await stream1.update(updatedContent, null, { anchor: false, publish: true })
-      await TestUtils.delay(500)
-
-      expect(feed1.length).toEqual(2) // create + update
-      expect(feed1[0].eventType).toBe(EventType.INIT)
-      expect(feed1[1].eventType).toBe(EventType.DATA)
-      expect(feed2.length).toEqual(2) //load + pubsub update
-      expect(feed2[0].eventType).toBe(EventType.INIT)
-      expect(feed2[1].eventType).toBe(EventType.DATA)
-      expect(feed2[0].content.test).toBe(content.test)
-      expect(feed2[0].commitId).toStrictEqual(feed1[0].commitId)
-      // test pubsub propagating the update from stream1 being inside the feed
-      expect(feed2[1].commitId).toStrictEqual(feed1[1].commitId)
-      expect(feed2[1].metadata).toStrictEqual(feed1[1].metadata)
-      expect(feed2[1].content).toStrictEqual(updatedContent)
-
-      s1.unsubscribe()
-      s2.unsubscribe()
-    }, 20000)
-
-    test('add entry after anchoring stream', async () => {
-      const feed: FeedDocument[] = []
-      const s = ceramic1.feed.aggregation.documents.subscribe((s) => {
-        feed.push(s)
+    const abortController = new AbortController()
+    const doneStreaming = readable1
+      .pipeTo(writable1, { signal: abortController.signal })
+      .catch(() => {
+        // Ignore, as it is an Abort Signal
       })
+    const model = await Model.create(ceramic1, modelDefinition())
+    const document = await ModelInstanceDocument.create(
+      ceramic1,
+      { myData: 0 },
+      { model: model.id }
+    )
+    await document.replace({ myData: 1 }, undefined, { anchor: false })
+
+    await TestUtils.waitForConditionOrTimeout(async () => emissions.length === 3, 5000)
+
+    // Model.create
+    expect(emissions[0].commitId).toEqual(model.commitId)
+    // document.create
+    expect(emissions[1].commitId).toEqual(document.allCommitIds[0])
+    // document.replace
+    expect(emissions[2].commitId).toEqual(document.allCommitIds[1])
+    abortController.abort()
+    await doneStreaming
+  })
 
-      const stream = await TileDocument.create(
-        ceramic1,
-        { hello: `world-${Math.random()}` },
-        null,
-        {
-          anchor: true,
-        }
-      )
-      const state$ = await ceramic1.repository.load(stream.id, {})
-      // request anchor
-      await ceramic1.repository.anchor(state$, {})
-      // process anchor
-      await CoreUtils.anchorUpdate(ceramic1, stream)
-
-      expect(feed.length).toEqual(2) // 2 commits = 1 genesis commit + 1 anchor commit
-      expect(feed[0].content).toEqual(feed[1].content)
-      expect(feed[0].metadata).toEqual(feed[1].metadata)
-      expect(feed[0].commitId.equals(feed[1].commitId)).toEqual(false)
-      expect(feed[0].eventType).toBe(EventType.INIT)
-      expect(feed[1].eventType).toBe(EventType.TIME)
-      s.unsubscribe()
+  testIfV3('add entry after anchoring stream', async () => {
```

Comment:
what's the reason this test can't work with rust-ceramic / in recon mode?  It's looking like most (all?) of the users of data feed api are going to be building on rust-ceramic, so seems like we'd really want all the data feed tests to work with rust-ceramic

Reply:
No reason. It is a mistake.